src/components/Onboarding/Onboarding.tsx

```diff
@@ -1,75 +1,94 @@
 import Page1 from './OnboardingPage1';
 import Page2 from './OnboardingPage2';
 import Page3 from './OnboardingPage3';
 import Page4 from './OnboardingPage4';
+import {
+  createHome,
+  createHomeOwner,
+  updateHome,
+} from '../../graphql/mutations';
+import { CreateHomeInput } from '../../API';
 
-const Onboarding: React.FC = () => {
-  const [page, setPage] = useState('page1');
-  const [homeData, setHomeData] = useState({
-    AC: '',
-    EV: '',
-    address: '',
-    bathrooms: '',
-    bedrooms: '',
-    city: '',
-    electricEnergyUsage: '',
-    fuelType: '',
-    gasUsage: '',
-    heaterAge: '',
-    hotTub: '',
-    outdoor: '',
-    pool: '',
-    squareFeet: '',
-    state: '',
-    waterFuelType: '',
-    waterHeaterAge: '',
-    waterUse: '',
-    year: '',
-    zipCode: '',
-  });
+const Stack = createStackNavigator();
+
+interface Props {
+  homeInformation?: CreateHomeInput;
+}
+
+const Onboarding: React.FC<Props> = ({ homeInformation }) => {
+  const [page, setPage] = useState('');
+  const [homeData, setHomeData] = useState(
+    homeInformation !== undefined ? homeInformation : homeInfo,
+  );
   const { setAppState } = useContext(AppContext);
 
   useEffect(() => {
+    const addHome = async () => {
+      let result;
+      if (homeInformation !== undefined) {
+        result = await API.graphql({
+          query: updateHome,
+          variables: { input: homeData },
+        });
+      } else {
+        result = await API.graphql({
+          query: createHome,
+          variables: { input: homeData },
+        });
+      }
+
+      await API.graphql({
+        query: createHomeOwner,
+        variables: {
+          input: {
+            homeID: result.data.createHome.id,
+            homeOwnerID: result.data.createHome.owner,
+          },
+        },
+      });
+    };
+
     if (page === 'submit') {
-      // submit home info to backend here
-      console.log(homeData);
+      addHome();
       setAppState('App');
     }
-  }, [page, homeData, setAppState]);
+  }, [page, homeData, setAppState, homeInformation]);
 
   return (
-    <>
-      {page === 'page1' && (
-        <Page1
-          setPage={setPage}
-          setHomeData={setHomeData}
-          homeData={homeData}
-        />
-      )}
-      {page === 'page2' && (
-        <Page2
-          setPage={setPage}
-          setHomeData={setHomeData}
-          homeData={homeData}
-        />
-      )}
-      {page === 'page3' && (
-        <Page3
-          setPage={setPage}
-          setHomeData={setHomeData}
-          homeData={homeData}
-        />
-      )}
-      {page === 'page4' && (
-        <Page4
-          setPage={setPage}
-          setHomeData={setHomeData}
-          homeData={homeData}
-        />
-      )}
-    </>
+    <Stack.Navigator>
+      <Stack.Screen
+        name="Page 1"
+        children={() => <Page1 setHomeData={setHomeData} homeData={homeData} />} // eslint-disable-line react/no-children-prop
```

Comment:
Based on [this](https://reactnavigation.org/docs/screen#children), it seems the `component` prop would achieve the same purpose as `children` here, without needing to pass an arrow function. I think component would make this a bit more readable, but this is nitpicky, so I'll approve regardless of whether you include this or not.

Reply:
I have to use the children since I need to pass props here. The component prop does not support passing props to the component itself

Reply:
yep, you're right. I confused the two - `children` is necessary here, not `component`.